lib/fog/vsphere/models/compute/cdrom.rb

```diff
@@ -0,0 +1,83 @@
+module Fog
+  module Compute
+    class Vsphere
+      class Cdrom < Fog::Model
+        attribute :filename
+        attribute :name
+        attribute :key
+        attribute :server_id
+        attribute :controller_key
+        attribute :unit_number
+        attribute :start_connected
+        attribute :allow_guest_control
+        attribute :connected
+
+        def initialize(attributes={} )
+          # Assign server first to prevent race condition with persisted?
+          self.server_id = attributes.delete(:server_id)
+
+          super defaults.merge(attributes)
+        end
+
+        def to_s
+          name
+        end
+
+        def destroy
+          requires :server_id, :key, :unit_number
+
+          service.destroy_vm_cdrom(self)
+          true
+        end
+
+        def save
+          requires :server_id
+
+          if unit_number.nil?
+            used_unit_numbers = server.cdroms.map { |cdrom| cdrom.unit_number }
+            max_unit_number = used_unit_numbers.max
+
+            if max_unit_number > server.cdroms.size
+              # If the max ID exceeds the number of cdroms, there must be a hole in the range. Find a hole and use it.
+              self.unit_number = max_unit_number.times.to_a.find { |i| used_unit_numbers.exclude?(i) }
+            else
+              self.unit_number = max_unit_number + 1
+            end
+          else
+            if server.cdroms.any? { |cdrom| cdrom.unit_number == self.unit_number && cdrom.id != self.id }
+              raise "A cdrom already exists with that unit_number, so we can't save the new cdrom"
+            end
+          end
+
+          data = service.add_vm_cdrom(self)
+
+          if data['task_state'] == 'success'
+            # We have to query vSphere to get the cdrom attributes since the task handle doesn't include that info.
+            created = server.cdroms.all.find { |cdrom| cdrom.unit_number == self.unit_number }
+
+            self.key = created.key
+            self.filename = created.filename
+            self.unit_number = created.unit_number
+
+            true
+          else
+            false
+          end
+        end
+
+        def server
+          requires :server_id
+          service.servers.get(server_id)
+        end
```

Comment:
Yeap. Could you test it out?

I'm pretty sure it works just like xenserver and because of that it will work.

This method will define a getter that return an object instead of the id only.